test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1072,6 +1072,54 @@ UCS_TEST_P(test_ucp_am_nbx_align, multi)
     test_am_send_recv(fragment_size() * 5, 0, 0, UCP_AM_FLAG_PERSISTENT_DATA);
 }
 
+UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx_align)
+
+class test_ucp_am_nbx_align_offset : public test_ucp_am_nbx_align {
+public:
+    test_ucp_am_nbx_align_offset() : test_ucp_am_nbx_align()
+    {
+        m_align_offset = ucs::rand() % 100;
+    }
+
+    virtual ucp_worker_params_t get_worker_params() override
+    {
+        auto params = test_ucp_am_nbx_align::get_worker_params();
+
+        params.field_mask     |= UCP_WORKER_PARAM_FIELD_AM_ALIGN_OFFSET;
+        params.am_align_offset = m_align_offset;
+        return params;
+    }
+
+    virtual ucs_status_t
+    am_data_handler(const void *header, size_t header_length, void *data,
+                    size_t length, const ucp_am_recv_param_t *rx_param) override
+    {
+        test_ucp_am_nbx::am_data_handler(header, header_length, data, length,
+                                         rx_param);
+        if (rx_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_DATA) {
+            auto aligned_offset = (uintptr_t)data + m_align_offset;
+            EXPECT_EQ(0u, aligned_offset % m_alignment) << " data ptr " << data;
+        }
+
+        return UCS_OK;
+    }
+
+private:
+    size_t m_align_offset;
+};
+
+UCS_TEST_P(test_ucp_am_nbx_align_offset, basic)
+{
+    test_am_send_recv(fragment_size() / 2, 0, 0, UCP_AM_FLAG_PERSISTENT_DATA);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_align_offset, multi)
+{
+    test_am_send_recv(fragment_size() * 5, 0, 0, UCP_AM_FLAG_PERSISTENT_DATA);
+}
+
+UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx_align_offset)
+
```

Comment:
i'd also test that asking for some unreasonable big offset (bigger than the expected data) does not provoke any errors.

Reply:
Fixed